testing/bdd/features/validation/3350-value-types.feature

```diff
@@ -0,0 +1,156 @@
+    When it has extent attribute with value <extent>
+    Then document is valid
+
+    Examples:
+    | xml_file             | extent  |
+    | 3350_value_types.xml | 1% 1%   |
+    | 3350_value_types.xml | 1c 1c   |
+    | 3350_value_types.xml | 1px 1px |
+
+  Scenario: Invalid extent values
+    Given an xml file <xml_file>
+    When it has extent attribute with value <extent>
+    Then document is invalid
+
+    Examples:
+    | xml_file             | extent |
+    | 3350_value_types.xml | 1 1    |
+    | 3350_value_types.xml | -1c 1c |
+    | 3350_value_types.xml | 1px    |
+
+  # SPEC-CONFORMANCE: R95
+  Scenario: Valid font size values
+    Given an xml file <xml_file>
+    When it has tts:fontSize attribute with value <font_size>
+    Then document is valid
+
+    Examples:
+    | xml_file             | font_size |
+    | 3350_value_types.xml | 1%  2c    |
+    | 3350_value_types.xml | 1.5px     |
+    | 3350_value_types.xml | 1c 0c     |
+    | 3350_value_types.xml | -1%  2c   |
+    | 3350_value_types.xml | +1px      |
+
+
+  Scenario: Invalid font size values
+    Given an xml file <xml_file>
+    When it has tts:fontSize attribute with value <font_size>
+    Then document is invalid
+
+    Examples:
+    | xml_file             | font_size |
+    | 3350_value_types.xml | 1% 1% 1%  |
+    | 3350_value_types.xml | 1em       |
+    | 3350_value_types.xml | 1c1c      |
+
+
+  # SPEC-CONFORMANCE: R97
+  Scenario: Valid line padding values
+    Given an xml file <xml_file>
+    When it has linePadding attribute with value <line_padding>
+    Then document is valid
+
+    Examples:
+    | xml_file             | line_padding |
+    | 3350_value_types.xml | 1c           |
+    | 3350_value_types.xml | 0.5c         |
+    | 3350_value_types.xml | .5c         |
+
+  Scenario: Invalid line padding values
+    Given an xml file <xml_file>
+    When it has linePadding attribute with value <line_padding>
+    Then document is invalid
+
+    Examples:
+    | xml_file             | line_padding |
+    | 3350_value_types.xml | 1%           |
+    | 3350_value_types.xml | 1px          |
+    | 3350_value_types.xml | -1c          |
+    | 3350_value_types.xml | 1em          |
+
+  # SPEC-CONFORMANCE: R98
+  Scenario: Valid line height values
+    Given an xml file <xml_file>
+    When it has lineHeight attribute with value <line_height>
+    Then document is valid
+
+    Examples:
+    | xml_file             | line_height |
+    | 3350_value_types.xml | normal      |
+    | 3350_value_types.xml | 1.5%        |
+    | 3350_value_types.xml | 1c          |
+    | 3350_value_types.xml | 1px         |
+
+  Scenario: Invalid line height values
+    Given an xml file <xml_file>
+    When it has lineHeight attribute with value <line_height>
+    Then document is invalid
+
+    Examples:
+    | xml_file             | line_height |
+    | 3350_value_types.xml | hello       |
+    | 3350_value_types.xml | 1em         |
+    | 3350_value_types.xml | 1c 2c       |
```

Comment:
Okay, same problem with the lengthType here, because the XSD was built following the syntax and the >=0 part was missed. I can make a regex specific for lineHeightType to remove the negative part.